src/core/timer.c

```diff
@@ -169,6 +170,48 @@ static int timer_setup_persistent(Timer *t) {
         return 0;
 }
 
+static int timer_set_fixed_delay(Timer *t) {
+        struct prefix {
+                sd_id128_t machine_id;
+                bool is_system_manager;
+                uid_t uid;
+        } *prefix;
+
+        static const uint8_t hash_key[] = {
+                0x51, 0x0a, 0xdb, 0x76, 0x29, 0x51, 0x42, 0xc2,
+                0x80, 0x35, 0xea, 0xe6, 0x8e, 0x3a, 0x37, 0xbd
+        };
+
+        _cleanup_free_ uint8_t *fixed_delay_data = NULL;
+        size_t dlen, id_strlen;
+        uint64_t hash;
+        int r;
+
+        assert(t);
+
+        id_strlen = strlen(UNIT(t)->id);
+        dlen = sizeof(struct prefix) + id_strlen;
+
+        fixed_delay_data = new0(uint8_t, dlen);
+        if (!fixed_delay_data)
+                return -ENOMEM;
+
+        prefix = (struct prefix *) fixed_delay_data;
+
+        r = sd_id128_get_machine(&prefix->machine_id);
+        if (r < 0)
+                return r;
+
+        prefix->is_system_manager = MANAGER_IS_SYSTEM(UNIT(t)->manager);
+        prefix->uid = getuid();
+        mempcpy(fixed_delay_data + sizeof(struct prefix), UNIT(t)->id, id_strlen);
+
+        hash = siphash24(fixed_delay_data, dlen, hash_key);
+        t->fixed_delay_usec = hash % t->random_usec;
+
+        return 0;
```

Comment:
oh, no need to put together a structure like this, you can use siphash24_init() followed by four siphash24_compress() for the four pieces of information to hash, and finished by a siphash24_finalize() to get the result. That works without memory allocation, and is what we usually do.